services/QuillLMS/app/views/blog_posts/show.html.erb

```diff
@@ -4,8 +4,9 @@
   <% active_tab = 'All resources'%>
 <% end %>
 
-<%= render partial: 'teachers/shared/scorebook_tabs' if current_user %>
-<%= render partial: 'pages/shared/teacher_center_navbar', locals: {active_tab: active_tab} %>
+<%= render partial: 'teachers/shared/scorebook_tabs' if current_user && current_user.role != 'student' %>
+<%= render partial: 'pages/shared/teacher_center_navbar', locals: {active_tab: active_tab} if (!current_user || current_user&.role == 'teacher') %>
```

Comment:
I think you can remove the safe navigation on this one: `(!current_user || current_user.role == 'teacher')`